tests/nlu/test_components.py

```diff
@@ -220,3 +222,144 @@ async def test_validate_component_keys_raises_warning_on_invalid_key(
         )
 
     assert "You have provided an invalid key" in record[0].message.args[0]
+
+
+@pytest.mark.parametrize(
+    "pipeline_template,should_warn",
+    [
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "CRFEntityExtractor"},
+                {"name": "DIETClassifier"},
+            ],
+            True,
+        ),
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "DIETClassifier"},
+            ],
+            False,
+        ),
+    ],
+)
+def test_warn_of_competing_extractors(pipeline_template, should_warn):
+    config = RasaNLUModelConfig({"pipeline": pipeline_template})
+    trainer = Trainer(config)
+
+    if should_warn:
+        with pytest.warns(UserWarning):
+            rasa.nlu.components.warn_of_competing_extractors(trainer.pipeline)
+    else:
+        with pytest.warns(None) as records:
+            rasa.nlu.components.warn_of_competing_extractors(trainer.pipeline)
+
+        assert len(records) == 0
```

Comment:
optional: I'd favor to have to separate test cases of this instead of `if ... else` for the asserts.

Reply:
Would make the code a bit simpler, but you would add duplicate some more lines. Don't have a strong preference here

Reply:
In my opinion fine to have duplicate code in tests and rather make them self contained and concise. However, fine for me.